File: src/main/java/io/zeitwert/ddd/aggregate/model/enums/CodeAggregateType.java
package io.zeitwert.ddd.aggregate.model.enums;

import io.zeitwert.ddd.enums.model.base.EnumeratedBase;
import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
public final class CodeAggregateType extends EnumeratedBase {

	@Override
	public boolean equals(Object other) {
		return super.equals(other);
	}

	@Override
	public int hashCode() {
		return super.hashCode();
	}

}

File: fm-dddrive/src/main/java/dddrive/ddd/model/enums/CodeAggregateType.kt
package dddrive.ddd.model.enums

import dddrive.ddd.model.Enumerated

class CodeAggregateType(
	override val id: String,
	override val defaultName: String,
) : Enumerated {

	override val enumeration: CodeAggregateTypeEnum
		get() = CodeAggregateTypeEnum.instance

}
